File: CS/Solution2.Module/MyLogonParameters.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;
using System.ComponentModel;
using DevExpress.ExpressApp.Security;

namespace Solution2.Module {
    [NonPersistent]
    public class MyLogonParameters : AuthenticationStandardLogonParameters {
        private SecurityType securityType = SecurityType.Complex;
        public SecurityType SecurityType {
            get {
                return securityType;
            }
            set {
                securityType = value;
            }
        }
    }
    public enum SecurityType {
        Simple = 0,
        Complex = 1
    }
}

File: VB/Solution2.Module/MyLogonParameters.vb
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Security

Namespace Solution2.Module

    <NonPersistent>
    Public Class MyLogonParameters
        Inherits AuthenticationStandardLogonParameters

        Private securityTypeField As SecurityType = SecurityType.Complex

        Public Property SecurityType As SecurityType
            Get
                Return securityTypeField
            End Get

            Set(ByVal value As SecurityType)
                securityTypeField = value
            End Set
        End Property
    End Class

    Public Enum SecurityType
        Simple = 0
        Complex = 1
    End Enum
End Namespace
